src/main/java/io/cryostat/recordings/RecordingTargetHelper.java

```diff
@@ -251,9 +250,13 @@ public Future<Optional<InputStream>> getRecording(
         return future;
     }
 
-    public Future<Void> deleteRecording(
-            ConnectionDescriptor connectionDescriptor, String recordingName) {
-        return this.deleteRecording(connectionDescriptor, recordingName, true);
+    public Future<Void> deleteRecording(ConnectionDescriptor connectionDescriptor, String recordingName) {
+        Matcher m = SNAPSHOT_NAME_PATTERN.matcher(recordingName);
+        if (m.matches()) {
+            return this.deleteRecording(connectionDescriptor, recordingName, false, true);
+        } else {
+            return this.deleteRecording(connectionDescriptor, recordingName, true, true);
+        }
```

Comment:
Since the only difference here is the snapshot regex check and the boolean value passed into the call to the private method, does it make sense to just perform that check within the private method and remove the parameter from the method signature?